### PR TITLE
An idea of supporting writeCloser

### PR DESCRIPTION
**Pull request by the-plate**
We could do something like this to support upload files(bytes) via WriteCloser... 
Any hints or recommendations are welcome.

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -449,3 +450,22 @@ func (c *Client) WriteStream(path string, stream io.Reader, _ os.FileMode) (err
 		return newPathError("WriteStream", path, s)
 	}
 }
+
+func (c *Client) WriteWrCls(_ context.Context, path string, _ os.FileMode) (writer io.WriteCloser, err error) {
+	pr, pw := io.Pipe()
+	done := make(chan error, 1)
+
+	go func() {
+		defer pr.Close()
```
Comment (chripo) [inline client.go:459]: could lead to multiple close calls due the RoundTripper

File: client.go [hunk 2]
```diff
@@ -449,3 +450,22 @@ func (c *Client) WriteStream(path string, stream io.Reader, _ os.FileMode) (err
...
+
+		_, err := c.put(path, pr)
```
Comment (chripo) [inline client.go:461]: `put` want create parent collections

File: client.go [hunk 3]
```diff
@@ -449,3 +450,22 @@ func (c *Client) WriteStream(path string, stream io.Reader, _ os.FileMode) (err
 		return newPathError("WriteStream", path, s)
 	}
 }
+
+func (c *Client) WriteWrCls(_ context.Context, path string, _ os.FileMode) (writer io.WriteCloser, err error) {
```
Comment (chripo) [inline client.go:454]: 1. the naming could be improved
2. why is the context required if not used?
3. is it necessary to declare the err?, it`s always `nil`

**Review by chripo — COMMENTED** [review]
some test would be required too.

**Comment by chripo** [comment]
Please rebase on `master` and try to address the feedback.
thank you!

**Comment by the-plate** [comment]
I will clone the pull request, the idea is not needed any more...

**Pull request closed without merging**